app/src/main/java/com/example/toyproject/ui/today/TodaySkinFragment.kt

```diff
@@ -0,0 +1,100 @@
+package com.example.toyproject.ui.today
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.isVisible
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
+import com.example.toyproject.base.BaseFragment
+import com.example.toyproject.data.db.entities.Skins
+import com.example.toyproject.databinding.FragmentTodaySkinBinding
+import com.example.toyproject.util.Resource
+import com.example.toyproject.util.extensions.bindAdapterTransform
+import com.example.toyproject.util.extensions.bindImage
+import com.example.toyproject.util.extensions.setAdapter
+import com.example.toyproject.viewmodel.SkinViewModel
+import com.irozon.alertview.AlertActionStyle
+import com.irozon.alertview.AlertStyle
+import com.irozon.alertview.AlertView
+import com.irozon.alertview.objects.AlertAction
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class TodaySkinFragment : BaseFragment<FragmentTodaySkinBinding>(),
+    TodaySkinAdapter.OnClickListener {
+
+    @Inject
+    lateinit var adapter: TodaySkinAdapter
+
+    private val skinViewModel: SkinViewModel by viewModels()
+
+    lateinit var skinTitle: String
+
+    override val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> FragmentTodaySkinBinding =
+        FragmentTodaySkinBinding::inflate
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        setupUI()
+        setUpObserver()
+    }
+
+    private fun setupUI() {
+        binding.apply {
+            setAdapter(dsView, adapter)
+            bindAdapterTransform(dsView, true)
+        }
+    }
+
+    private fun setUpObserver() {
+        skinViewModel.getAllSkins.observe(viewLifecycleOwner, Observer {
+            when (it) {
+                is Resource.Loading -> {
+                    showProgress(true)
+                    showRecyclerView(false)
+                }
+                is Resource.Success -> {
+                    showProgress(false)
+                    showRecyclerView(true)
+                    todayRandomSkin(it.data)
+                    adapter.submitList(it.data)
+                }
+                is Resource.Error -> {
+                    showProgress(false)
+                    binding.txtError.isVisible = true
+                }
+            }
+        })
+    }
+
+    private fun todayRandomSkin(skins: List<Skins>) {
+        skins.random().apply {
+            binding.apply {
+                txtSkinTitle.text = skinTitle
+                txtSkinKinds.text = skinKinds
+                bindImage(imgTodaySkin, imageUrl)
+            }
+        }
```

Comment:
```suggestion
        skins.random().apply {
            with(binding) {
                txtSkinTitle.text = skinTitle
                txtSkinKinds.text = skinKinds
                bindImage(imgTodaySkin, imageUrl)
            }
        }
```
제안드려봅니다 !

Reply:
더불어, 프로젝트 내에서 apply, with 를 쓰는 기준이 있으신 것 같습니다 🙂 
어떤 경우에 사용하시는 건지 궁금합니다~

Reply:
apply { } 스코프 안에 with 혹은 apply와 같은 다른 스코프 함수를 사용하는건 코드를 더 읽기 어렵게 하는것 같아요.

Reply:
범위 지정 함수 같은 경우 항상 헷갈리는 부분인데요. 정확하게 스코프를 구분하려 하진 않았는데, 구분한다면 78/79번 라인은 프로퍼티의 초기화인 apply를 사용하는게 맞다고 생각하고 80번 라인은 with를 사용하는게 맞다고 생각이 드네요. 코드를 좀 더 깔끔하게 만든다면 random() 함수가 어떤 값을 계산 하는것이니 아래와 같이 수정하는 게 어떨까 싶습니다.
```suggestion
         skins.random().run {
            binding.apply {
                txtSkinTitle.text = skinTitle
                txtSkinKinds.text = skinKinds
                bindImage(imgTodaySkin, imageUrl)
            }
        }
```